app/controllers/invitations_controller.rb

```diff
@@ -8,63 +9,71 @@ def new
+      participant = AssignmentParticipant.where('user_id =? and parent_id =?', user_invitee.id, student.parent_id).first
+      if participant && !team.full?
+        team_member = TeamsUser.where(['team_id =? and user_id =?', team.id, user_invitee.id])
         # check if invited user is already in the team
-        if !team_member.empty?
-          flash[:note] = "The user \"#{user.name}\" is already a member of the team."
-        else
+        if team_member.empty? && Invitation.is_invited?(student.user_id, user_invitee.id, student.parent_id)
           # check if the invited user is already invited (i.e. awaiting reply)
-          if Invitation.is_invited?(student.user_id, user.id, student.parent_id)
-            @invitation = Invitation.new
-            @invitation.to_id = user.id
-            @invitation.from_id = student.user_id
-            @invitation.assignment_id = student.parent_id
-            @invitation.reply_status = 'W'
-            @invitation.save
-          else
-            flash[:note] = "You have already sent an invitation to \"#{user.name}\"."
-          end
+          # reply_status is the current replying status
+          # W means there's a invitation reply waiting to be sent
+          # A means the invitation is accepted
+          # D means the invitaion is declined
+          @invitation = Invitation.new
+          @invitation.to_id = user_invitee.id
+          @invitation.from_id = student.user_id
+          @invitation.assignment_id = student.parent_id
+          @invitation.reply_status = 'W'
+          @invitation.save
         end
       end
     end
+    if !user_invitee
+      flash[:note] = "The invited user \"#{params[:user][:name].strip}\" does not exist. Please make sure the name entered is correct."
+    elsif !participant
+      flash[:note] = "The invited user \"#{params[:user][:name].strip}\" is not a participant of this assignment."
+    elsif team.full?
+      flash[:error] = "Your team already has the maximum number members."
+    elsif !team_member.empty?
+      flash[:note] = "The invited user \"#{user_invitee.name}\" is already a member of the team."
+    elsif Invitation.is_invited?(student.user_id, user_invitee.id, student.parent_id)
+      flash[:note] = "You have already sent an invitation to \"#{user_invitee.name}\"."
+    end
 
-    update_join_team_request user, student
+    update_join_team_request user_invitee, student
 
     redirect_to view_student_teams_path student_id: student.id
   end
 
-  def update_join_team_request(user, student)
+  # update the request to join team 
+  # if join team request exist, handle invitaion automatically
+  # used in above case at line 51 when the condition is fulfiled
+  def update_join_team_request(user_invitee, student)
     # update the status in the join_team_request to A
-    if user && student
-      participant = AssignmentParticipant.where(['user_id =? and parent_id =?', user.id, student.parent_id]).first
+    if user_invitee && student
```

Comment:
This is the student in the existing team, as in line 14, we didn't modify this name because it looked reasonable by then. Now that you point out it's confusing, I think it does mean that the old naming needs improving.